examples/src/main/scala/examples/hybrid/history/HybridHistory.scala

```diff
@@ -407,14 +417,25 @@ class HybridHistory(val storage: HistoryStorage,
     def in(m: HybridBlock): Boolean = loserChain.exists(s => s sameElements m.id)
 
     val winnerChain = chainBack(forkBlock, in, limit).get.map(_._2)
-    val i = loserChain.indexWhere(id => id sameElements winnerChain.head)
+    val i = loserChain.indexWhere { id =>
+      winnerChain.headOption match {
+        case None                  => false
+        case Some(winnerChainHead) => id sameElements winnerChainHead
+      }
+    }
     (winnerChain, loserChain.takeRight(loserChain.length - i))
-  }.ensuring(r => r._1.head sameElements r._2.head)
```

Comment:
Here, for example, both the original code and the refactored code will throw an exception when `r._1.head` or `r._2.head` is undefined. In the original code, this will occur because `head` is undefined. In the refactored code this will happen later because the condition will evaluate to false and `ensuring` will throw an exception. So, both codes achieve the same effect, but the original code is:

1. simpler and more concise
2. more efficient (throwing the exception earlier)
3. easier to understand for readers who don't know `Option`, `for` notation and `getOrElse`.